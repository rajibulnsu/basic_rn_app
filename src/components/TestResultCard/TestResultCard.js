import React from 'react';
import { View, Text, ProgressBarAndroid } from 'react-native';
import styles from './TestResultCard.style';

const TestResultCard = props => {
  const { biomarker, color, percentage } = props.item;

  return (
    <View style={styles.cardLayout}>
      <Text style={styles.title}>Biomarker: {biomarker}</Text>
      <Text style={styles.text}>Percentage: {percentage}%</Text>
      <ProgressBarAndroid
        style={styles.percentageBar}
        styleAttr="Horizontal"
        color={color}
        indeterminate={false}
        progress={percentage / 100}
        />
    </View>
  );
};

export default TestResultCard;
