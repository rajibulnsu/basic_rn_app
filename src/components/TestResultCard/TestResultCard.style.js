import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  cardLayout: {
    maxWidth: '100%',
    minHeight: 90,
    margin: 10,
    padding: 20,
    borderRadius: 5,
    shadowColor: 'black',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.25,
    shadowRadius: 5,
    elevation: 5,
    backgroundColor: 'white',
  },
  title: {
    fontSize: 20,
  },
  text: {
    fontSize: 16,
  },
  percentageBar: {
    padding: 10,
    transform: [{ scaleX: 0.7 }, { scaleY: 5 }],
  },
});

export default styles;
