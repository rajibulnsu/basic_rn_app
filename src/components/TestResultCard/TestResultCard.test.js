import React from 'react';
import renderer from 'react-test-renderer';

import TestResultCard from './TestResultCard';

describe('TestResultCard', () => {
  it('should match the snapshot', () => {
    const item = {
      biomarker: 'biomarker',
      color: '#ddd',
      percentage: 0.25,
    };
    const testResultCard = renderer.create(<TestResultCard item={item}/>);

    expect(testResultCard).toMatchSnapshot();
  });
});
