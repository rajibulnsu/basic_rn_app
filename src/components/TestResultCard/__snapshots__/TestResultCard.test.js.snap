// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestResultCard should match the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 5,
      "elevation": 5,
      "margin": 10,
      "maxWidth": "100%",
      "minHeight": 90,
      "padding": 20,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 5,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
      }
    }
  >
    Biomarker: 
    biomarker
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
      }
    }
  >
    Percentage: 
    0.25
    %
  </Text>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
          "borderColor": "red",
          "borderWidth": 1,
        },
        Object {
          "padding": 10,
          "transform": Array [
            Object {
              "scaleX": 0.7,
            },
            Object {
              "scaleY": 5,
            },
          ],
        },
      ]
    }
  />
</View>
`;
